$NetBSD: patch-src_couch_rebar.config.script,v 1.3 2022/11/29 19:16:04 jperkin Exp $

Set pkgsrc standard paths.
Use pkg-config to set SpiderMonkey CFLAGS/LDFLAGS.

--- src/couch/rebar.config.script.orig	2022-04-06 03:31:40.000000000 +0000
+++ src/couch/rebar.config.script
@@ -112,6 +112,11 @@ ProperConfig = case code:lib_dir(proper)
     _ -> [{d, 'WITH_PROPER'}]
 end.
 
+CFLAGS = [os:getenv("CFLAGS")],
+LDFLAGS = [os:getenv("LDFLAGS")],
+JSFLAGS = string:strip(os:cmd("pkg-config --cflags mozjs185"), right, $\n),
+JSLIBS = string:strip(os:cmd("pkg-config --libs mozjs185"), right, $\n),
+
 {JS_CFLAGS, JS_LDFLAGS} = case os:type() of
     {win32, _} when SMVsn == "1.8.5" ->
         {
@@ -120,8 +125,8 @@ end.
         };
     {unix, _} when SMVsn == "1.8.5" ->
         {
-            "-DXP_UNIX -I/usr/include/js -I/usr/local/include/js",
-            "-L/usr/local/lib -lmozjs185 -lm"
+            "$CFLAGS -DXP_UNIX " ++ JSFLAGS,
+            "$LDFLAGS -lm " ++ JSLIBS
         };
     {win32, _} when SMVsn == "60" ->
         {
@@ -187,8 +192,8 @@ CouchJSEnv = case SMVsn of
         ]
 end.
 
-IcuEnv = [{"DRV_CFLAGS",  "$DRV_CFLAGS -DPIC -O2 -fno-common"},
-          {"DRV_LDFLAGS", "$DRV_LDFLAGS -lm -licuuc -licudata -licui18n -lpthread"}].
+IcuEnv = [{"DRV_CFLAGS",  "$CFLAGS $DRV_CFLAGS -DPIC -O2 -fno-common"},
+          {"DRV_LDFLAGS", "$LDFLAGS $DRV_LDFLAGS -lm -licuuc -licudata -licui18n -lpthread"}].
 IcuDarwinEnv = [{"CFLAGS", "-DXP_UNIX -I/usr/local/opt/icu4c/include -I/opt/homebrew/opt/icu4c/include"},
                 {"LDFLAGS", "-L/usr/local/opt/icu4c/lib -L/opt/homebrew/opt/icu4c/lib"}].
 IcuBsdEnv = [{"CFLAGS", "-DXP_UNIX -I/usr/local/include"},
@@ -203,9 +208,7 @@ BaseSpecs = [
         %% couchjs
         {".*", CouchJSPath, CouchJSSrc, [{env, CouchJSEnv}]},
         % ejson_compare
-        {"darwin", ComparePath, CompareSrc, [{env, IcuEnv ++ IcuDarwinEnv}]},
-        {"linux",  ComparePath, CompareSrc, [{env, IcuEnv}]},
-        {"bsd",   ComparePath, CompareSrc, [{env, IcuEnv ++ IcuBsdEnv}]},
+        {".*",  ComparePath, CompareSrc, [{env, IcuEnv}]},
         {"win32",  ComparePath, CompareSrc, [{env, IcuWinEnv}]}
 ].
 
